lib/adhearsion/call_controller.rb

```
@@ -108,8 +108,9 @@ def exec(controller = self)
 
     def bg_exec(completion_callback = nil)
       Celluloid::ThreadHandle.new(Celluloid.actor_system) do
-        catching_standard_errors do
+        begin
           exec_with_callback completion_callback
+        rescue StandardError
```

Are there any cases where an error would be silently swallowed here? I understand that there's a place deeper in the code which logs it, but are there any ways for an exception to happen outside that catcher/logger?

---

True, it could happen. `exec_with_callback` and `exec` are short and simple but the optional `completion_callback` wouldn't be covered either. Any suggestions?

---

What about removing the deeper exception catcher/logger, and leaving this one in place? Any downside?

---

I guess we'd keep the `on_error` call and the reraise, removing just the trigger. I've checked and `execute!` is not called from elsewhere. It's also marked (but is not really) private so this should be fine. I'll give it a try.